### PR TITLE
Feature response

### PR DESCRIPTION
**Pull request by Raymundogs**

### CODE REVIEW

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java [hunk 1]
```diff
@@ -26,24 +29,58 @@
...
     public Docket api() {
         return new Docket(DocumentationType.SWAGGER_2)
+                .groupName("API")
+                .securitySchemes(Arrays.asList(apiKey()))
                 .select()
                 .paths(PathSelectors.ant("/api/*"))
                 .apis(RequestHandlerSelectors.basePackage("org.fundacion.jala.converter"))
                 .build()
                 .apiInfo(apiDetails());
     }
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: These lines could try to be up to 120 characters

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java [hunk 2]
```diff
@@ -26,24 +29,58 @@
...
+    @Bean
+    public Docket apiAuthentication() {
+        return new Docket(DocumentationType.SWAGGER_2)
+                .groupName("AUTHENTICATION")
+                .select()
+                .paths(PathSelectors.ant("/authenticate"))
+                .apis(RequestHandlerSelectors.basePackage("org.fundacion.jala.converter"))
+                .build()
+                .apiInfo(apiDetails());
+    }
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: These lines could try to be up to 120 characters

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java [hunk 3]
```diff
@@ -26,24 +29,58 @@
...
+    @Bean
+    public Docket apiRegistration() {
+        return new Docket(DocumentationType.SWAGGER_2)
+                .groupName("REGISTRATION")
+                .select()
+                .paths(PathSelectors.ant("/register"))
+                .apis(RequestHandlerSelectors.basePackage("org.fundacion.jala.converter"))
+                .build()
+                .apiInfo(apiDetails());
+    }
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: These lines could try to be up to 120 characters

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java [hunk 4]
```diff
@@ -26,24 +29,58 @@
...
     /**
-     * Creates a Docket.
+     * Creates the API Docket.
      *
      * @return a Docket with Swagger configuration.
      */
     @Bean
     public Docket api() {
         return new Docket(DocumentationType.SWAGGER_2)
+                .groupName("API")
```
Comment (JRojas999) [inline src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: It should be in a line until 120 characteres.

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java [hunk 5]
```diff
@@ -63,4 +100,22 @@ private ApiInfo apiDetails() {
...
+     */
+    private ApiKey apiKey() {
+        return new ApiKey("JWT", "Authorization", "header");
+    }
+
+    /**
+     * hello
+     * @return
+     */
+    @Bean(name = "multipartResolver")
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: this javadoc is not in format

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java [hunk 6]
```diff
@@ -63,4 +100,22 @@ private ApiInfo apiDetails() {
...
+     * Creates an API key.
+     *
+     * @return the API key.
+     */
+    private ApiKey apiKey() {
+        return new ApiKey("JWT", "Authorization", "header");
+    }
+
+    /**
+     * hello
```
Comment (JRojas999) [inline src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: It should have a better description.

File: src/main/java/org/fundacion/jala/converter/controller/response/ErrorResponse.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    public ErrorResponse(final String initialStatus, final String initialMessage) {
+        super(initialStatus);
+        this.message = initialMessage;
+    }
+
+    /**
+     * Gets the response's message.
+     * @return a String with the message.
+     */
+    public String getMessage() {
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/response/ErrorResponse.java]: the javadoc must have a space between the description and the return

File: src/main/java/org/fundacion/jala/converter/controller/response/ErrorResponse.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+        return message;
+    }
+
+    /**
+     * Sets the response's message.
+     * @param newMessage a String to change the message.
+     */
+    public void setMessage(final String newMessage) {
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/response/ErrorResponse.java]: the javadoc must have a space between the description and the param

File: src/main/java/org/fundacion/jala/converter/controller/response/PaoPaoResponse.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+     */
+    public String getStatus() {
+        return status;
+    }
+
+    /**
+     * Sets the response status.
+     * @param newStatus a String to change the status.
+     */
+    public void setStatus(final String newStatus) {
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/response/PaoPaoResponse.java]: the javadoc must have a space between the description and the return and param

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java [hunk 10]
```diff
@@ -63,4 +100,22 @@ private ApiInfo apiDetails() {
...
+     * @return
```
Comment (JRojas999) [inline src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: It should be have a blank line between desciption and @return.

File: src/main/java/org/fundacion/jala/converter/view/controllers/ClientRequest.java [hunk 11]
```diff
@@ -33,6 +38,8 @@
 import java.io.BufferedInputStream;
 import java.net.MalformedURLException;
 import java.net.URL;
+import java.util.Properties;
+
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/view/controllers/ClientRequest.java]: This space should be removed

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java [hunk 12]
```diff
@@ -26,24 +29,58 @@
...
                 .paths(PathSelectors.ant("/api/*"))
                 .apis(RequestHandlerSelectors.basePackage("org.fundacion.jala.converter"))
                 .build()
                 .apiInfo(apiDetails());
     }
 
+    /**
+     * Creates the authentication Docket.
+     *
+     * @return a Docket with Swagger configuration
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: A point "." is needed at the end of the sentence

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java [hunk 13]
```diff
@@ -26,24 +29,58 @@
...
+                .paths(PathSelectors.ant("/authenticate"))
+                .apis(RequestHandlerSelectors.basePackage("org.fundacion.jala.converter"))
+                .build()
+                .apiInfo(apiDetails());
+    }
+
+    /**
+     * Creates the registration Docket.
+     *
+     * @return a Docket with Swagger configuration
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: A point "." is needed at the end of the sentence

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java [hunk 14]
```diff
@@ -26,24 +29,58 @@
...
+                .apis(RequestHandlerSelectors.basePackage("org.fundacion.jala.converter"))
+                .build()
+                .apiInfo(apiDetails());
+    }
+
     /**
      * Customizes the API Information.
      *
-     * @return the ApiInfo with the customization values.
+     * @return the ApiInfo with the custom values
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: A point "." is needed at the end of the sentence

**Review by JRojas999 — APPROVED** [review]

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java (hunk 1 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: solved it.

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java (hunk 2 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: solved it.

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java (hunk 3 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: solved it.

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java (hunk 4 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: solved it.

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java (hunk 5 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: corrected it.

File: src/main/java/org/fundacion/jala/converter/controller/response/PaoPaoResponse.java [hunk 15]
```diff
@@ -1,18 +1,14 @@
...
  *
- * @author Jessicka Moya Andrade
+ * @author Raymundo Guaraguara Sansusty
  */
-package org.fundacion.jala.converter.response;
+package org.fundacion.jala.converter.controller.response;
 
-/**
- * This class creates the application's responses.
- */
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/response/PaoPaoResponse.java]: javadoc missing from class

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java (hunk 6 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: corrected it.

File: src/main/java/org/fundacion/jala/converter/controller/response/ErrorResponse.java (hunk 7 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/controller/response/ErrorResponse.java]: added it.

File: src/main/java/org/fundacion/jala/converter/controller/response/ErrorResponse.java (hunk 8 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/controller/response/ErrorResponse.java]: added it.

File: src/main/java/org/fundacion/jala/converter/controller/response/SuccessAuthenticationResponse.java [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+ * Copyright (c) 2021 Fundacion Jala.
+ * This software is the confidential and proprietary information of Fundacion Jala
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * license agreement you entered into with Fundacion Jala
+ *
+ * @author Raymundo Guaraguara Sansusty
+ */
+package org.fundacion.jala.converter.controller.response;
+
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/response/SuccessAuthenticationResponse.java:12]: javadoc missing from class

File: src/main/java/org/fundacion/jala/converter/controller/response/SuccessRegistrationResponse.java [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+ * Copyright (c) 2021 Fundacion Jala.
+ * This software is the confidential and proprietary information of Fundacion Jala
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * license agreement you entered into with Fundacion Jala
+ *
+ * @author Raymundo Guaraguara Sansusty
+ */
+package org.fundacion.jala.converter.controller.response;
+
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/response/SuccessRegistrationResponse.java:12]: javadoc missing from class

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java (hunk 10 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: corrected it.

File: src/main/java/org/fundacion/jala/converter/controller/response/PaoPaoResponse.java (hunk 9 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/controller/response/PaoPaoResponse.java]: added it.

File: src/main/java/org/fundacion/jala/converter/view/controllers/ClientRequest.java (hunk 11 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/view/controllers/ClientRequest.java]: removed it.

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java (hunk 12 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: added it.

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java (hunk 13 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: added it.

File: src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java (hunk 14 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/controller/config/SwaggerConfig.java]: added it.

File: src/main/java/org/fundacion/jala/converter/controller/response/PaoPaoResponse.java (hunk 15 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/controller/response/PaoPaoResponse.java]: added it.

File: src/main/java/org/fundacion/jala/converter/controller/response/SuccessAuthenticationResponse.java (hunk 16 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/controller/response/SuccessAuthenticationResponse.java:12]: added it.

File: src/main/java/org/fundacion/jala/converter/controller/response/SuccessRegistrationResponse.java (hunk 17 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/controller/response/SuccessRegistrationResponse.java:12]: added it.

File: src/main/java/org/fundacion/jala/converter/controller/response/ErrorResponse.java [hunk 18]
```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2021 Fundacion Jala.
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/response/ErrorResponse.java:2]: after copyright must have a space

File: src/main/java/org/fundacion/jala/converter/controller/response/PaoPaoResponse.java [hunk 19]
```diff
@@ -1,17 +1,16 @@
 /**
  * Copyright (c) 2021 Fundacion Jala.
- *
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/response/PaoPaoResponse.java]: after copyright must have a space

File: src/main/java/org/fundacion/jala/converter/controller/response/SuccessAuthenticationResponse.java [hunk 20]
```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2021 Fundacion Jala.
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/response/SuccessAuthenticationResponse.java:2]: after copyright must have a space

File: src/main/java/org/fundacion/jala/converter/controller/response/SuccessRegistrationResponse.java [hunk 21]
```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2021 Fundacion Jala.
```
Comment (jpgonzales27) [inline src/main/java/org/fundacion/jala/converter/controller/response/SuccessRegistrationResponse.java:2]: after copyright must have a space

**Review by jpgonzales27 — APPROVED** [review]

**Review by edsonarios — APPROVED** [review]

**Pull request merged**